cmd/swarm/swarm-smoke/pss.go

```diff
@@ -0,0 +1,237 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha1"
+	"fmt"
+	"math/rand"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/metrics"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/ethereum/go-ethereum/swarm/pss"
+	"github.com/ethereum/go-ethereum/swarm/testutil"
+
+	cli "gopkg.in/urfave/cli.v1"
+)
+
+const (
+	pssModeRaw = iota
+	pssModeAsym
+	pssModeSym
+)
+
+type pssJob struct {
+	msg  []byte
+	mode int
+	prox bool
+}
+
+type pssNode struct {
+	hostIdx        int
+	addr           []byte
+	pubkey         string
+	client         *rpc.Client
+	deregisterFunc func()
+	msgC           chan pss.APIMsg
+}
+
+// keyed by hex representation of overlay address
+type pssSession struct {
+	topic        pss.Topic
+	msgC         chan pss.APIMsg
+	nodes        []*pssNode
+	requiredJobs map[string]*pssJob
+	allowedJobs  map[string]*pssJob
+}
+
+// send messages using pss asymetric encryption across random nodes
+func pssAsymCheck(ctx *cli.Context, tuid string) error {
+	// use input seed if it has been set
+	if inputSeed != 0 {
+		seed = inputSeed
+	}
+	rand.Seed(int64(seed))
+	if pssMessageCount <= 0 {
+		pssMessageCount = 1
+		log.Warn("message count should be a positive number. Defaulting to %d", pssMessageCount)
```

Comment:
This should be structured log line - odd number of arguments, for example:
`log.Warn(fmt.Sprintf("message count should be positive. defaulting to %v", pssMessageCount))` in order to have 1 argument.